enclave_build/src/utils.rs

```diff
@@ -0,0 +1,431 @@
+// Copyright 2019-2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+use oci_distribution::client::{ImageLayer};
+use oci_distribution::client::{ImageData};
+
+use serde_json::Value;
+use std::env;
+use std::io::Read;
+
+#[derive(Debug, PartialEq)]
+#[allow(clippy::enum_variant_names)]
+pub enum ExtractError {
+    ImageError(String),
+    ManifestError(String),
+    LayerError(String),
+    ConfigError(String),
+    EnvCmdError(String),
+    ImageHashError(String),
+}
+
+/// Contains the logic for the extraction of image data from an ImageData struct to be stored later
+/// in the local cache
+/// 
+/// The ImageData struct represents the data of an image and is what the pull operation from
+/// https://github.com/krustlet/oci-distribution returns
+pub struct ExtractLogic {}
+
+impl ExtractLogic {
```

Comment:
This looks like the code in extract.rs, just wrapped wrapped in an impl? Is there something I'm missing here? Why duplicate this code?